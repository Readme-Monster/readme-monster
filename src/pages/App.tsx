import { ToastContainer } from "react-toastify";
import { getAuth, onAuthStateChanged } from "firebase/auth";
import React, { useEffect, useState, useContext } from "react";
import { Routes } from "./Routes";
import { app } from "../firebaseApp";
import ThemeContext from "context/ThemeContext";
import LoadingSpinner from "../components/Common/LoadingSpinner/LoadingSpinner";
import "react-toastify/dist/ReactToastify.css";

function App() {
  const auth = getAuth(app);
  const context = useContext(ThemeContext);
  const [init, setInit] = useState<boolean>(false);
  const [isAuthenticated, setIsAuthenticated] = useState<boolean>(!!auth?.currentUser);

  useEffect(() => {
    onAuthStateChanged(auth, user => {
      if (user) {
        setIsAuthenticated(true);
      } else {
        setIsAuthenticated(false);
      }
      setInit(true);
    });
  }, [auth]);

  return (
    <>
      <div className={context.theme === "light" ? "white" : "dark"}>
        <ToastContainer 
          position="top-center" 
          autoClose={5000}
          hideProgressBar={false}
          newestOnTop={false}
          closeOnClick
          rtl={false}
          pauseOnFocusLoss
          draggable
          pauseOnHover
          theme={context.theme === "light" ? "white" : "dark"}
        />
        {init ? <Routes isAuthenticated={isAuthenticated} /> : <LoadingSpinner/>}
      </div>
    </>
  );
}

export default App;
