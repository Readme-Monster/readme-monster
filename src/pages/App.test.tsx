import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "./App";

describe("App Component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
