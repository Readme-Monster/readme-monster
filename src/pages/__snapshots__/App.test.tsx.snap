// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div
        class="stiky z-10 w-full h-12 text-white top-0"
      >
        <div
          class="flex w-full h-12 relative justify-between items-center my-0 mx-auto py-0 px-4"
        >
          <div
            class="h-full flex justify-center items-center"
          >
            <a
              href="/"
            >
              <img
                alt="logo"
                class="h-8 w-8"
                src="/images/rm-logo.png"
              />
            </a>
          </div>
          <div
            class="flex justify-center items-center"
          >
            <a
              href="https://github.com/"
            >
              <img
                alt="github"
                class="h-8 w-8 mr-3"
                src="/images/github-logo.svg"
              />
            </a>
            <img
              alt="dark"
              class="h-8 w-8 mr-3"
              src="/images/light-to-dark.svg"
            />
            <button
              class="p-0 btn btn-transparent"
              type="button"
            >
              <img
                alt="mypage"
                class="h-8 w-8"
                src="/images/mypage.svg"
              />
            </button>
          </div>
        </div>
      </div>
      <div>
        Homepage
      </div>
    </div>
  </div>
</DocumentFragment>
`;
